lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -4722,6 +4722,59 @@ FIRRTLType RefSubOp::inferReturnType(ValueRange operands,
       loc, "ref.sub op requires a RefType of vector or bundle base type");
 }
 
+//===----------------------------------------------------------------------===//
+// Optional Group Operations
+//===----------------------------------------------------------------------===//
+
+LogicalResult GroupOp::verify() {
+  auto groupName = getGroupName();
+  auto *parentOp = (*this)->getParentOp();
+
+  auto nestedReferences = groupName.getNestedReferences();
+  if (nestedReferences.empty()) {
+    if (!isa<FModuleOp>(parentOp)) {
+      auto diag = mlir::emitError(getLoc())
```

Comment:
Can this use `emitOpError` instead of manually inserting operation name, specifying location?

Reply:
This is a habit of forgetting that you can negate the default behavior of dumping the whole op (which may be massive for groups) with `-emit-op-on-diagnostic=false`. I'll switch to the more canonical `emitOpError`.